src/lib/common/Kconfig

```diff
@@ -81,11 +81,43 @@ config DEFAULT_LOG_LEVEL_DEBUG
 	bool "debug"
 endchoice
 
-config KDBUS
-	bool "Kdbus"
-	depends on SOL_PLATFORM_LINUX && HAVE_SYSTEMD && HAVE_KDBUS
+config SOL_BUS
+	bool "Sol-bus"
+	depends on SOL_PLATFORM_LINUX && HAVE_SYSTEMD && HAVE_SD_BUS
 	default n
 
+menu "Hardware support"
+config HARDWARE_PWM
+	bool "pwm"
+	help
+          Globaly enables pwm hardware support.
+	default y
+
+config HARDWARE_SPI
+	bool "spi"
+	help
+          Globaly enables spi hardware support.
+	default y
+
+config HARDWARE_UART
+	bool "uart"
+	help
+          Globaly enables uart hardware support.
+	default y
+
+config HARDWARE_I2C
+	bool "i2c"
+	help
+          Globaly enables i2c hardware support.
+	default y
+
+config HARDWARE_GPIO
+	bool "gpio"
+	help
+          Globaly enables gpio hardware support.
+	default y
+endmenu
+
```

Comment:
we're using HAVE_ for a different purpose and I wouldn't like to mix them all. HAVE_* always indicates what your host system provides and these HAVE_* variables are defined by the dependency-resolver, it doesn't tell you what soletta has but what the host system provides us.

About the kconfig docs. Where did you see this suggestion? this is totally up to the users to define the patterns they want to follow, if you see the different kconfig users you'll identify multiple patterns. If you're talking about kconfig-language.txt or their friends they're indicating/suggesting common uses for linux kernel itself, since we're not the linux kernel we can use whatever we want. Remember, the building context/environment for linux kernel is totally different to ours, they don't link against shared libraries for example, and they know very little about the host system.